From 3a6e636fb3bed4bda1a737658411a2af43c86e00 Mon Sep 17 00:00:00 2001
From: Keith Miller <keith_miller@apple.com>
Date: Wed, 22 May 2024 08:38:40 -0700
Subject: [PATCH] Fix linker error in JSC testing configuration
 https://bugs.webkit.org/show_bug.cgi?id=274523 rdar://128542072

Reviewed by Mark Lam.

It seems like someone holding a Ref/RefPtr to UnlinkedMetadataTable didn't
include the inlines header. Rather than figure out how to find the missing
include I just moved the destructor to the .cpp file. I don't think the
destructor is hot enough that inlining is needed.

* Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp:
(JSC::UnlinkedMetadataTable::~UnlinkedMetadataTable):
* Source/JavaScriptCore/bytecode/UnlinkedMetadataTableInlines.h:
(JSC::UnlinkedMetadataTable::~UnlinkedMetadataTable): Deleted.

Canonical link: https://commits.webkit.org/279126@main
---
 Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp   | 7 +++++++
 .../JavaScriptCore/bytecode/UnlinkedMetadataTableInlines.h | 7 -------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp b/Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp
index bc59830c5e59..e2f02566db6d 100644
--- a/Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp
+++ b/Source/JavaScriptCore/bytecode/UnlinkedMetadataTable.cpp
@@ -170,4 +170,11 @@ void UnlinkedMetadataTable::finalize()
     }
 }
 
+UnlinkedMetadataTable::~UnlinkedMetadataTable()
+{
+    ASSERT(!m_isLinked);
+    if (m_hasMetadata)
+        MetadataTableMalloc::free(m_rawBuffer);
+}
+
 } // namespace JSC
diff --git a/Source/JavaScriptCore/bytecode/UnlinkedMetadataTableInlines.h b/Source/JavaScriptCore/bytecode/UnlinkedMetadataTableInlines.h
index 5fec54dbdfba..a52be2e6aec6 100644
--- a/Source/JavaScriptCore/bytecode/UnlinkedMetadataTableInlines.h
+++ b/Source/JavaScriptCore/bytecode/UnlinkedMetadataTableInlines.h
@@ -60,13 +60,6 @@ ALWAYS_INLINE UnlinkedMetadataTable::UnlinkedMetadataTable(EmptyTag)
 {
 }
 
-ALWAYS_INLINE UnlinkedMetadataTable::~UnlinkedMetadataTable()
-{
-    ASSERT(!m_isLinked);
-    if (m_hasMetadata)
-        MetadataTableMalloc::free(m_rawBuffer);
-}
-
 ALWAYS_INLINE unsigned UnlinkedMetadataTable::addEntry(OpcodeID opcodeID)
 {
     ASSERT(!m_isFinalized && opcodeID < s_offsetTableEntries - 1);
-- 
2.46.0

